### PR TITLE
DEBUG Planner2389

### PR DESCRIPTION
**Pull request by aloubyansky**
### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

**Comment by triceo** [comment]
ok to test.

**Comment by triceo** [comment]
Thank you for looking into this, @aloubyansky !
If you don't mind, I converted this PR to a draft, so that it doesn't show up on people's radars until it is ready.

**Comment by aloubyansky** [comment]
That makes sense, thanks!

**Comment by rsynek** [comment]
ok to test

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 1]
```diff
@@ -127,7 +127,9 @@
         <maven.jdbc.driver.jar/>
         <maven.jdbc.username/>
         <maven.jdbc.password/>
-        <maven.jdbc.url>jdbc:h2:</maven.jdbc.url>
+        <!-- Relative path in H2 JDBC URL must be explicit.
+             See http://www.h2database.com/javadoc/org/h2/api/ErrorCode.html#c90011. -->
+        <maven.jdbc.url>jdbc:h2:tcp://localhost/./target/persistence-test</maven.jdbc.url>
```
Comment (ge0ffrey) [inline optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:132]: This is only for QA testing with an actual db, right?

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java [hunk 2]
```diff
@@ -33,14 +45,195 @@
...
 public class OptaPlannerProcessorConstraintProviderAndDrlTest {
 
     private static final String CONSTRAINTS_DRL = "customConstraints.drl";
 
     @RegisterExtension
     static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setBeforeAllCustomizer(new Runnable() {
+                @Override
+                public void run() {
+                    System.out.println("BEFORE ALL CUSTOMIZER");
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java:68]: Don't do System.out.println. Use logging instead.

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml (hunk 1 above)
Comment (aloubyansky) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:132]: I did not introduce that change, afaik, btw.

**Comment by aloubyansky** [comment]
I'm done with it.

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java [hunk 3]
```diff
@@ -33,14 +45,195 @@
...
+                                .setCurrentProject(currentProject)
+                                .build();
+                        System.out
+                                .println("  " + resolver
+                                        .resolve(new DefaultArtifact(currentProject.getGroupId(),
+                                                currentProject.getArtifactId(), "jar", currentProject.getVersion()))
+                                        .getArtifact().getFile());
+                    } catch (BootstrapMavenException e) {
+                        // TODO Auto-generated catch block
+                        e.printStackTrace();
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java:233]: don't eat exceptions

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java [hunk 4]
```diff
@@ -33,14 +45,195 @@
...
+                    }
+                }
+            })
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/constraints/OptaPlannerProcessorConstraintProviderAndDrlTest.java:236]: Do we really need all this code? Less is more.
Do we really need to parse the maven metadata and start manipulating files?

**Comment by rsynek** [comment]
@ge0ffrey this PR was opened to investigate an issue that manifests only in the Jenkins build and blocks merging the https://github.com/kiegroup/optaplanner/pull/1254, which now waits for Quarkus 1.13.3.Final to become available.

**Comment by ge0ffrey** [comment]
> I'm done with it.

@ aloubyansky I hope you didn't get demotivated by our review process - or my code-oriented (not person-oriented) tone, don't take that personally. We do want to merge in your improvements and those of this PR, just without introducing new liabilities (such as System.out, regardless if your or someone else made that change, its part of this PR, so part of this review).

**Comment by ge0ffrey** [comment]
Ah, nevermind me then trying to review this PR - I didn't catch on this was a pure DEBUG PR, not one to actually merge. Thanks for looking into this @aloubyansky and @rsynek

**Comment by aloubyansky** [comment]
Right, this code was never meant to be actually merged. Sorry for the confusion.

**Comment by ge0ffrey** [comment]
No worries, thanks for your help :)

**Pull request closed without merging**